core/src/com/unciv/logic/map/TileInfo.kt

```diff
@@ -1195,15 +1210,69 @@ open class TileInfo : IsPartOfGameInfoSerialization {
     /** Sets tile improvement to pillaged (without prior checks for validity)
      *  and ensures that matching [UniqueType.CreatesOneImprovement] queued buildings are removed. */
     fun setPillaged() {
+        if (getPillagableImprovement() == null && getUnpillagedRoad() == RoadStatus.None)
+            return
         // http://well-of-souls.com/civ/civ5_improvements.html says that naval improvements are destroyed upon pillage
         //    and I can't find any other sources so I'll go with that
         if (isLand) {
             // Setting turnsToImprovement might interfere with UniqueType.CreatesOneImprovement
             removeCreatesOneImprovementMarker()
-            improvementInProgress = improvement
-            turnsToImprovement = 2
+            improvementInProgress = null  // remove any in progress work as well
+            turnsToImprovement = 0
+            // if no Repair action, destroy improvements instead
+            if (ruleset.tileImprovements[Constants.repair] == null) {
+                if (getPillagableImprovement() != null)
+                    improvement = null
+                else
+                    roadStatus = RoadStatus.None
+            } else {
+                // otherwise use pillage/repair systems
+                if (getPillagableImprovement() != null) {
+                    improvementIsPillaged = true
+                } else {
+                    roadIsPillaged = true
+                }
+            }
+        } else {
+            improvement = null
```

Comment:
Putting this at the top as 'if (!isLand) { improvement=null; return } would simplify the logic

Reply:
Done